### ISSUE TITLE
Chef run failure in build_bins.sh while building VM cluster

### ISSUE THREAD

**Issue by vt0r** [bloomberg/chef-bach#848]
When running `tests/automated_install.sh`, the `build_bins.sh` script is executed, and though no error output is displayed during the execution of `build_bins.sh`, a stacktrace is printed (below) to `/var/chef/cache/chef-stacktrace.out`, which causes the overall bootstrap setup procedure to fail later on when this file is detected. Error message will follow shortly when I can reproduce it again.

The error can be bypassed until this issue is resolved using this terrible hack:
```diff
--- a/bootstrap_chef.sh
+++ b/bootstrap_chef.sh
@@ -85,6 +85,8 @@ fi
 $SSH_CMD "tar -xzf /chef-bcpc-host/cluster.tgz -C /home/vagrant" || (echo '### Vagrant ssh failed to deploy cluster environment returned $? ###'; e
 echo "Building bins"
 $SSH_CMD "cd $BCPC_DIR && sudo ./build_bins.sh"
+echo "HACK - removing stacktrace file, since the build_bins run succeeded."
+$SSH_CMD "sudo rm -f /var/chef/cache/chef-stacktrace.out"
 echo "Setting up chef server"
 $SSH_CMD "cd $BCPC_DIR && sudo ./setup_chef_server.sh ${CHEF_ENVIRONMENT}"
 echo "Setting up chef cookbooks"
```

**Comment by cbaenziger** [comment]
For reference, while I see the following when my `build_bins.sh` run kicks off:
```[2017-05-11T05:39:56+00:00] INFO: HTTP Request Returned 404 Not Found: Object not found: chefzero://localhost:8889/nodes/bcpc-bootstrap.bcpc.example.com
```

I can not replicate having a `/var/chef/cache/chef-stacktrace.out` by the type my PXE install kicks off. Are you sure your `build_bins.sh` Chef run is where you are getting this stack trace?

**Comment by pu239ppy** [comment]
Ignoring this blows up during C-A-R
 * mysql_database[pdns] action create

````    ============================================================================
====
    Error executing action `create` on resource 'mysql_database[pdns]'
    ============================================================================
====
````

**Comment by vt0r** [comment]
@cbaenziger - Yes. I wrapped the following line in debug output (`sudo cat /var/chef/cache/chef-stacktrace.out`), and it doesn't exist before the script runs, but the stacktrace is present afterwards (L87):

```bash
$SSH_CMD "cd $BCPC_DIR && sudo ./build_bins.sh"
```

@pu239ppy - That is really just letting it get to the next error :) I created #847 to describe that issue.

**Comment by vt0r** [comment]
Also worth noting I do not see this issue when I copied the built bins dir from the bootstrap to a hypervisor's chef-bach directory. Will debug this further once I'm able to make more progress in the project I'm working on.

**Comment by vt0r** [comment]
Fixed in #882 with a hack. Since this would be difficult to troubleshoot and likely not worth the time to remove a small hack, I'm closing this issue.
